### PR TITLE
Add users list to admin dashboard

### LINKED ISSUE #598: Add users list to dashboard
@satra requested an easily accessible list of users available for inviting users to the dandi slack.
Add a list of all current users to the dashboard, sorted by join date descending, that can easily be copy/pasted.

### PR DESCRIPTION
**Pull request by dchiquito**
Fixes #598

### CODE REVIEW

File: dandiapi/api/dashboard.py [hunk 1]
```diff
@@ -59,6 +62,9 @@ def _non_valid_assets(self):
     def _uploads(self):
         return Upload.objects.annotate()
 
+    def _users(self):
+        return User.objects.select_related('metadata').order_by('-date_joined')
```
Comment (satra) [inline dandiapi/api/dashboard.py:66]: is this restricted to active users?

Comment (dchiquito) [reply dandiapi/api/dashboard.py:66]: No, but I just added a column to also show the current status. Would you rather unapproved users not show up at all?

Comment (satra) [reply dandiapi/api/dashboard.py:66]: yes. at least for my present needs, only approved users.

File: dandiapi/api/templates/dashboard/index.html [hunk 2]
```diff
@@ -111,5 +111,41 @@ <h1>Non-valid Assets</h1>
...
+        <td>{{ user.date_joined }}</td>
+        <td><a href="{% url 'user-approval' user.username %}">{{ user.metadata.status }}</a></td>
+      </tr>
+      {% endfor %}
+    </table>
+  </div>
+
 </div>
-{% endblock %}
+{% endblock %}
```
Comment (mvandenburgh) [inline dandiapi/api/templates/dashboard/index.html:151]:
```suggestion
{% endblock %}

```

**Review by mvandenburgh — APPROVED** [review]
Code LGTM

**Review by waxlamp — COMMENTED** [review]
Looks good to me. Deferring to Mike's reviews as binding.

**Pull request merged**